ivc/src/lib.rs

```diff
@@ -1,3 +1,116 @@
+//!
+//! The accumulation of the challenges, the homogeneous value and the blinders
+//! are done trivially as they are scalar field values.
+//! After that, the verifier has to perform foreign field ellictic curve
+//! additions and scalar multiplications `(r T0)`, `(r^2 T1)`, `(r^3 E2)` and
+//! `(r (C_i)_R)` for each column.
+//!
+//! First, we decide to defer the computations of
+//! the scalar multiplications, and we reduce the verifier work to compute only
+//! the foreign field elliptic curve additions. Therefore, the verifier has
+//! access already to the
+//! result of `r T0`, `r^2 T1`, `r^3 E2` and `r (C_i)_R`, and must only perform the
+//! addition. We call the commitments `r T0` the "scaled" commitment to `T0`, and
+//! the same for the others. The commitments `(C_i)_L`, `(C_i)_R` and `(C_i)_O` are
+//! called the "instance commitments".
+//!
+//! To perform foreign field elliptic curve addition, we split the commitments
+//! into 17 chunks of 15 bits and use additive lookups as described in
+//! [kimchi_msm::logup]. These 17 chunks will be used later to compute the
+//! scalar multiplications using a variant of the scheme described in [the MSM
+//! RFC](https://github.com/o1-labs/rfcs/blob/main/0013-efficient-msms-for-non-native-pickles-verification.md)
+//!
+//! The first component of our circuit is a hash function.
+//! We decided to use the Poseidon hash function, and implemented a generic one
+//! using the generic interpreter in [crate::poseidon_8_56_5_3_2]. The Poseidon
+//! instance we decided to go with is the traditional full/partial rounds. For a
+//! security of 128 bits, a substitution box of 5 and a state of 3 elements, we
+//! need 8 full rounds and 56 partial rounds for the scalar field BN254.
+//! Therefore, we can "absorb" 2 field elements per row (the third element is
+//! kept as a buffer in the Sponge construction).
+//!
+//! The cost of a single Poseidon hash in terms of constraints is 432
+//! constraints, and requires 435 columns, in addition to 192 round constants
+//! considered as constants "selectors" in our constraints.
+//! Note that having 432 constraints per Poseidon hash means that we must also
+//! have 432 constraints to accumulate the challenges used to combine the
+//! constraints. It is worth noting that these alphas are the same for all
+//! poseidon hashes.
+//! Combining constraints is done by using another selector, on a single row,
+//! see below (TODO).
+//! Note that the columns used by the poseidon hash must also be aggregated
+//! while running the IVC.
+//!
+//! The Poseidon hash is used to absorb all the instance commitments, the
+//! challenges and the scaled commitments.
+//! In our circuit, we first start by absorbing all the elliptic curve points. A
+//! particularity is that we will use different Poseidon instances for each
+//! "side" of the addition. For each point, we do assume (*at the moment*) that
+//! each point is encoded as 2 field elements in the field of the circuit
```

Comment:
In this paragraph (and a bit in the previous one, and in the following one) we are being overly specific with parameters and layouts that we might change next week. Having these numbers in the column file and only general structure of the module in this comment would minimise maintenance overhead.